### ISSUE TITLE
Auto-materialize not working for daily partitioned asset with end_offset=-1

### ISSUE THREAD

**Issue by mmutso-boku** [dagster-io/dagster#14692]
### Dagster version

1.3.7

### What's the issue?

A daily partitioned asset with end_offset=-1 will not get materialized by the auto-materialize daemon (or the asset reconc sensor), even when upstream asset's partition is materialized. The daemon will materialize correctly once when turning it on, but for the next partitions that appear, no runs will be started.

### What did you expect to happen?

Auto-materialize daemon materializes the asset with end_offset correctly.

### How to reproduce?

```
@asset(
    key_prefix="smth",
    group_name='group1',
    description='Upstream',
    compute_kind='Tracker',
    partitions_def=DailyPartitionsDefinition(datetime(2023, 1, 1))
)
def asset_a():
    return

@asset(partitions_def=DailyPartitionsDefinition(datetime(2022, 1, 1), end_offset=-1),
       compute_kind='sql',
       key_prefix=['smth2'],
       group_name='group1',
       description='Downstream',
       non_argument_deps={'smth/asset_a'},
       auto_materialize_policy=AutoMaterializePolicy.eager())
def asset_b(context):
    doStuff(context)
```

With the actual date of reproduction being 08.06, this is how I reproduced in local dev env:

1. set date to 05
2. start dagster
3. materialize asset_a 04 partition
4. shut down dagster
5. set date to 06
6. start dagster
7. materialize asset_a 05 partition
8. turn on auto-materialize daemon
9. asset_b 04 partition will get materialized
10. shut down dagster
11. set date to 07
12. start dagster
13. materialize asset_a 06 partition
14. asset_b 05 partition should now start, but it does not. 

The same problem also exists with the old asset_reconciliation sensor, and the same steps can be used to reproduce the issue, just instead of turning on the auto-materialize daemon, you turn on the asset reconciliation sensor. With the asset reconciliation sensor, if you wipe the cursor after step 14, it will "rediscover" that asset_b 05 partition should be materialized, and it will start a run for it, but when time comes for the next partition, the same problem scenario happens.

The problem appears in an actual environment deployed with helm in k8s, the steps above are for reproducing quickly in local dev env.

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
@OwenKephart - I suspect this is because of `time_window_partition_scope_minutes`? Is there a recommended value for that parameter that would address this problem?

**Comment by OwenKephart** [comment]
Hi @mmutso-boku ! Based on those exact symptoms, I believe that this issue may have been due to an error in the partition mapping logic, which incorrectly mapped partitions when using TimeWindowPartitionMappings with different values for `start_offset` and `end_offset`. This error was fixed in `dagster==1.1.13` -- do you mind confirming if you're still experiencing this issue?

If it doesn't it's worth a try to increase `time_window_partition_scope_minutes` (which represents the maximum age of a time partition that the sensor will consider) with something like:
```python
my_auto_materialize_policy = AutoMaterializePolicy.eager()._replace(
    time_window_partition_scope_minutes=2*24*60,
)
```
but I don't have high confidence that this will change the exact behavior based on your note:

> With the asset reconciliation sensor, if you wipe the cursor after step 14, it will "rediscover" that asset_b 05 partition should be materialized

as it seems like the sensor is able to materialize that partition, it's just not choosing to do so.

**Comment by sryza** [comment]
Closing this issue based on Owen's mention of a fix. Please reopen if still experiencing the issue.
